AirCasting/SessionViews/ChartViewModel.swift

```diff
@@ -0,0 +1,72 @@
+// Created by Lunar on 30/09/2021.
+//
+
+import Foundation
+import Charts
+
+final class ChartViewModel: ObservableObject {
+    @Published var entries: [ChartDataEntry] = []
+    
+    private let stream: MeasurementStreamEntity
+
+    private var timeUnit: Double {
+        stream.session.type == .mobile ? 60 : 60*60
+    }
+    
+    private var mainTimer: Timer?
+    private var firstTimer: Timer?
+    
+    deinit {
+        mainTimer?.invalidate()
+        firstTimer?.invalidate()
+    }
+    
+    init(stream: MeasurementStreamEntity) {
+        self.stream = stream
+        generateEntries()
+        startTimers(stream.session)
+    }
+    
+    private func startTimers(_ session: SessionEntity) {
+        let sessionStartTime = session.startTime!
+        let secondsUntilFullMinute = timeUnit - Double(Int(Date().timeIntervalSince(sessionStartTime)) % Int(timeUnit))
+
+        firstTimer = Timer.scheduledTimer(withTimeInterval: TimeInterval(secondsUntilFullMinute), repeats: false) { [weak self] timer in
+            self?.generateEntries()
+            self?.startMainTimer()
+        }
+    }
+    
+    func startMainTimer() {
+        mainTimer = Timer.scheduledTimer(withTimeInterval: timeUnit, repeats: true) { [weak self] timer in
+            self?.generateEntries()
+        }
+    }
+    
+    private func generateEntries() {
+        guard var intervalEnd = stream.lastMeasurementTime else {
+            return
+        }
+        var intervalStart = intervalEnd - timeUnit
+        
+        var entries = [ChartDataEntry]()
+        
+        for i in (0...8).reversed() {
```

Comment:
Could you please use here 'index' or 'number' instead of 'i'? I think it would be nicer to look at and could tell us more

Reply:
as for Janek's comment, I think i is generally used as an index var name so it's clear enough. However, I'd consider not hardcoding 8 here - make it a constant, so the chart is easily adjustable.